relationships/synthesis/APM-APPLICATION-to-INFRA-MSSQLINSTANCE.yml

```diff
@@ -6,7 +6,10 @@ relationships:
     conditions:
       - attribute: metricName
         # "datastore/instance/MSSQL/$host/$port"
-        startsWith: "datastore/instance/MSSQL/"
+        # ex: "datastore/instance/MSSQL/abc123.mydomain.local/1663" - keeps "abc123"
+        # ex: "datastore/instance/MSSQL/hostname/default"  - keeps "hostname"
+        regex: "datastore/instance/MSSQL\/([^\\.\\/]+).*/.*"
```

Comment:
According to the description of the PR, I don't think this change is going to produce the expected behaviour.
Adding a regex with a capturing group here doesn't make much sense since this regex is just a condition to trigger the rule but the value of capturing group is not going to be used at all.

I guess that you might want to use the regex with the capturing group when building the identifier of the target entity. If that's the case, you should change the those fragments that are using `attribute: metricName__4`.

`metricName__4` means the 4th segment of the metric name, being segments delimited by `/`. For the metric name `datastore/instance/MSSQL/hostname/default`, the value of `metricName__4` is `hostname`.

Although it's not document, [we do support regex as fragments of the identifier](https://github.com/newrelic/entity-definitions/blob/main/validator/schemas/relationship-synthesis-schema-v1.json#L368). It would be something like this:

```
identifier:
  fragments:
    - value: "ms-instance:"
    - capture:
      attribute: metricName  # or metricName__4 and "([^\\.\\/]+).*/.*"
      regex: "datastore/instance/MSSQL\/([^\\.\\/]+).*/.*"
```

Reply:
Ahhh, thank you @javimb ! First PR to this repo ✨ 

I was confused by not seeing a regex option in the documentation or any other examples in my brief look through other entity relationships; your tip here was helpful.